### PR TITLE
feat(tests): add GaugeReward unit tests

### PR DESCRIPTION
**Pull request by PierrickGT**

### CODE REVIEW

File: test/TokenVault.test.ts [hunk 1]
```diff
@@ -25,52 +21,52 @@ describe('Vault', () => {
...
 
     describe('increaseERC20Allowance()', () => {
         it('should allow owners to increase approval amount', async () => {
-            await vault.setApproved(wallet2.address, true)
-            await token.mock.allowance.withArgs(vault.address, wallet2.address).returns('0')
+            await tokenVault.setApproved(wallet2.address, true)
+            await token.mock.allowance.withArgs(tokenVault.address, wallet2.address).returns('0')
             await token.mock.approve.withArgs(wallet2.address, '1111').returns(true)
-            await vault.increaseERC20Allowance(token.address, wallet2.address, '1111')
+            await tokenVault.increaseERC20Allowance(token.address, wallet2.address, '1111')
```
Comment (asselstine) [inline test/TokenVault.test.ts:46]: You know what is interesting about this test- is that it doesn't assert the state change.

The mock doesn't have an expectation, so if the mock isn't called this test will still pass.

In this case, the easiest thing would be to instantiate a real token and check to see that the allowance was changed properly

File: test/TokenVault.test.ts [hunk 2]
```diff
@@ -25,52 +21,52 @@ describe('Vault', () => {
...
         it('should allow managers to increase approval amount', async () => {
-            await vault.setManager(wallet3.address)
-            await vault.setApproved(wallet2.address, true)
-            await token.mock.allowance.withArgs(vault.address, wallet2.address).returns('0')
+            await tokenVault.setManager(wallet3.address)
+            await tokenVault.setApproved(wallet2.address, true)
+            await token.mock.allowance.withArgs(tokenVault.address, wallet2.address).returns('0')
             await token.mock.approve.withArgs(wallet2.address, '1111').returns(true)
-            await vault.connect(wallet3).increaseERC20Allowance(token.address, wallet2.address, '1111')
+            await tokenVault.connect(wallet3).increaseERC20Allowance(token.address, wallet2.address, '1111')
```
Comment (asselstine) [inline test/TokenVault.test.ts:54]: Likewise here

File: test/TokenVault.test.ts [hunk 3]
```diff
@@ -25,52 +21,52 @@ describe('Vault', () => {
...
         })
     })
 
     describe('decreaseERC20Allowance()', () => {
         it('should allow manager to decrease approval amount', async () => {
-            await token.mock.allowance.withArgs(vault.address, wallet2.address).returns('1111')
+            await token.mock.allowance.withArgs(tokenVault.address, wallet2.address).returns('1111')
             await token.mock.approve.withArgs(wallet2.address, '111').returns(true)
-            await vault.decreaseERC20Allowance(token.address, wallet2.address, '1000')
+            await tokenVault.decreaseERC20Allowance(token.address, wallet2.address, '1000')
```
Comment (asselstine) [inline test/TokenVault.test.ts:62]: and here

File: test/TokenVault.test.ts [hunk 4]
```diff
@@ -25,52 +21,52 @@ describe('Vault', () => {
...
         })
 
         it('should allow manager to decrease approval amount', async () => {
-            await vault.setManager(wallet3.address)
-            await token.mock.allowance.withArgs(vault.address, wallet2.address).returns('1111')
+            await tokenVault.setManager(wallet3.address)
+            await token.mock.allowance.withArgs(tokenVault.address, wallet2.address).returns('1111')
             await token.mock.approve.withArgs(wallet2.address, '0').returns(true)
-            await vault.connect(wallet3).decreaseERC20Allowance(token.address, wallet2.address, '1111')
+            await tokenVault.connect(wallet3).decreaseERC20Allowance(token.address, wallet2.address, '1111')
```
Comment (asselstine) [inline test/TokenVault.test.ts:69]: and here

File: test/TokenVault.test.ts [hunk 5]
```diff
@@ -25,52 +21,52 @@ describe('Vault', () => {
...
         })
```
Comment (asselstine) [inline test/TokenVault.test.ts:70]: how about some revert checks for non-owner/managers?

**Review by asselstine — CHANGES_REQUESTED** [review]
Looks great! Just some quibbles on tests

File: test/TokenVault.test.ts (hunk 1 above)
Comment (PierrickGT) [reply test/TokenVault.test.ts]: Done: https://github.com/pooltogether/v4-core/pull/272/commits/b9457454673816223dca65caaf45b6fa23e2b5a0

File: test/TokenVault.test.ts (hunk 2 above)
Comment (PierrickGT) [reply test/TokenVault.test.ts]: Done: https://github.com/pooltogether/v4-core/pull/272/commits/b9457454673816223dca65caaf45b6fa23e2b5a0

File: test/TokenVault.test.ts (hunk 3 above)
Comment (PierrickGT) [reply test/TokenVault.test.ts]: Done: https://github.com/pooltogether/v4-core/pull/272/commits/b9457454673816223dca65caaf45b6fa23e2b5a0

File: test/TokenVault.test.ts (hunk 4 above)
Comment (PierrickGT) [reply test/TokenVault.test.ts]: Done: https://github.com/pooltogether/v4-core/pull/272/commits/b9457454673816223dca65caaf45b6fa23e2b5a0

File: test/TokenVault.test.ts (hunk 5 above)
Comment (PierrickGT) [reply test/TokenVault.test.ts]: Done: https://github.com/pooltogether/v4-core/pull/272/commits/b9457454673816223dca65caaf45b6fa23e2b5a0

**Review by asselstine — APPROVED** [review]

**Pull request merged**